Fix iOS flashlight not turning on

Yousef-Diab opened the pull request:
- Added AVCaptureSession management for torch control on iOS
- Created capture session with device input and dummy output
- Added proper error handling for session creation and torch level setting
- Fallback to TorchMode.On when SetTorchModeLevel fails
- Updated Dispose method to properly clean up session resources

Addresses unoplatform/Uno.Gallery#1249

**GitHub Issue:** closes unoplatform/Uno.Gallery#1249
https://github.com/unoplatform/Uno.Gallery/issues/1249

## PR Type:

- 🐞 Bugfix


## What is the current behavior? 🤔

IOS Flashlight not working.

## What is the new behavior? 🚀

A working Flashlight with better success and error handling

## PR Checklist ✅

- [X] 📝 Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.
- [ ] 🧪 Added [Runtime tests, UI tests, or a manual test sample](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] 📚 Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] 🖼️ Validated PR `Screenshots Compare Test Run` results.
- [X] ❗ Contains **NO** breaking changes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/unoplatform/uno?pullRequest=21418) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/unoplatform/uno?pullRequest=21418) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/unoplatform/uno?pullRequest=21418) it.</sub>

jeromelaban commented on src/Uno.UWP/Devices/Lights/Lamp.iOS.cs:
```diff
@@ -1,110 +1,172 @@
+            _isEnabled = _brightness > 0;
+
+            // Create capture session for torch control
+            _captureSession = new AVCaptureSession();
```
Thanks for your contribution!

I am surprised that we need to make this change, considering other implementations like MAUI do not need this.

https://github.com/dotnet/maui/blob/02b2e65b5364180dc4fd56c809dcda0169efeac7/src/Essentials/src/Flashlight/Flashlight.ios.cs

Have you identified what's was the original issue?

Yousef-Diab replied:
It looks like a commit message format validation issue, will check it soon

Yousef-Diab committed:
fix(ios): flashlight not turning on in iOS devices

- Added AVCaptureSession management for torch control on iOS
- Created capture session with device input and dummy output
- Added proper error handling for session creation and torch level setting
- Fallback to TorchMode.On when SetTorchModeLevel fails
- Updated Dispose method to properly clean up session resources
- Enhanced error logging and debugging information

Addresses #19366

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21418/wasm-skia-net9/index.html

Yousef-Diab committed:
fix(ios): fix Lamp.iOS.cs Formatting

MartinZikmund commented:
@Yousef-Diab unfortunately I think the problem is not in the implementation of `Lamp` itself, but specifically in https://github.com/unoplatform/uno.gallery, because if you try the `Lamp` sample in the Samples app, you can see that in this context the flashlight already works as is, so no further changes to the implementation of the API should be needed. What needs to be investigated is why this API does not work in context of the Gallery app.
